### PR TITLE
[FE-4958] High Precision Timestamps

### PR DESCRIPTION
**Pull request by CaitW**
[Immerse PR](https://github.com/omnisci/mapd-immerse/pull/5534)

### CODE REVIEW

File: src/charts/data-manager.js [hunk 1]
```diff
@@ -245,5 +251,6 @@ export function getNearestDataPoint (_mouseX, _dataObject, _scales, _keyType) {
 
 function convertToDate (_date) {
   // hacks to handle invalid date like "0014-06-08T00:00:00.000Z"
-  return new Date(new Date(_date).toString())
+  // return new Date(new Date(_date).toString())
+  return new Date(_date)
 }
```
Comment (thomasoniii) [inline src/charts/data-manager.js]: Is this comment out of date? I'm unsure if the hack is persisting in the new code, and the commented out code should probably be dropped for clarity's sake.

File: src/charts/data-manager.js [hunk 2]
```diff
@@ -148,7 +150,11 @@ export function augmentData (_data, _keyType, _sortBy, _fillData, _stackOffset,
 
   const dataByKey = d3.nest()
     .key(getKey)
-    .entries(flatDataSorted)
+    .entries(flatDataSorted.map(d => ({
+      x: _keyType === "time" ? d.x.toISOString() : d.x,
+      y: d.y,
+      id: d[keys.ID]
```
[the remark this answers is not in the document]
Comment (int3h) [reply src/charts/data-manager.js]: Oh man, thank you for the thorough review. This snippet here turned out to be the source of a bug where the popup wasn't showing the chart values. Good eye, saved me lots of debugging work.

File: src/charts/data-manager.js [hunk 3]
```diff
@@ -109,8 +109,10 @@ export function augmentData (_data, _keyType, _sortBy, _fillData, _stackOffset,
 
   // get all unique keys
   let allKeys = []
-  dataBySeries.forEach(d => { allKeys = allKeys.concat(d.values) })
-  allKeys = allKeys.map(d => d.key)
+  dataBySeries.forEach(d => {
+    allKeys = allKeys.concat(d.values)
+  })
+  allKeys = allKeys.map(d => d[keys.KEY])
```
[the remark this answers is not in the document]
Comment (int3h) [reply src/charts/data-manager.js]: Changed back to match `master` :)

File: src/charts/data-manager.js (hunk 1 above)
Comment (int3h) [reply src/charts/data-manager.js]: Changed back to match `master`.

**Pull request merged**